src/app/(main)/EventPage.tsx

```diff
@@ -68,13 +70,85 @@ const Roster = ({participants, orgs, startTime}: {participants: Record<string, P
   )
 }
 
+const activityUpdateOptions = {
+  complete: { label: 'Complete', callback: (activity: Activity) => ActivityActions.complete(activity.id, new Date().getTime()) },
+  close: { label: 'Close', callback: (activity: Activity) => ActivityActions.closed(activity.id, new Date().getTime()) },
+  reactivate: { label: 'Reactivate', callback: (activity: Activity) => ActivityActions.reactivate(activity.id) }
+};
+
+const getActivityUpdateOption = (activity: Activity) => {
+  if (!isActive(activity)) { return activityUpdateOptions.reactivate; }
+  if (isComplete(activity)) { return activityUpdateOptions.close; }
+  return activityUpdateOptions.complete;
+}
+
+const ActivityStatusUpdater = ({activity}: {activity: Activity}) => {
+
+  const dispatch = useAppDispatch();
+  const action = getActivityUpdateOption(activity);
+  const [promptingActivityState, setPromptingActivityState] = useState<boolean>(false);
+
+  return (
+    <>
+      <Button variant="outlined" size="small" onClick={() => setPromptingActivityState(true)}>{action.label}</Button>
+
+      <Dialog open={promptingActivityState} onClose={() => setPromptingActivityState(false)}>
+        <DialogTitle>{action.label} event?</DialogTitle>
+        <DialogContent>
+          <DialogContentText>Only perform this action if you are authorized to do so.</DialogContentText>
+        </DialogContent>
+        <DialogActions>
+          <Button onClick={() => setPromptingActivityState(false)}>Cancel</Button>
+          <Button autoFocus onClick={() => {
+            dispatch(action.callback(activity));
+            setPromptingActivityState(false);
+            }}>{action.label}</Button>
+        </DialogActions>
+      </Dialog>
+    </>
+  );
+
+}
+
+const ActivityStatusSummaryFields = ({activity}:{activity: Activity}) => {
+  const participants = Object.values(activity.participants);
+
+  if (isComplete(activity)) {

```

Comment:
Why not show all the details when it's closed? Presumably at that point the only reason someone would be looking at a closed mission or event would be to examine all the details.